colobot-base/src/level/robotmain.cpp

```diff
@@ -4782,7 +4782,7 @@ bool CRobotMain::IOWriteScene(const std::filesystem::path& filename,
     if (!ostr.is_open()) return false;
 
     bool bError = false;
-    long version = 1;
+    long version = 2;
```

Comment:
This part of the code and similar version checks should be completely rewritten, but since it is outside of this PR's scope I'd suggest just leaving the 'version' as is for now.

Reply:
I do not understand. What do you mean by "completely rewritten"? Why?

Reply:
I'd rather not have magic numbers to keep track of and update before every single major release anywhere in the code; or version defines like this one for that matter: https://github.com/colobot/colobot/blob/c56d240a0f8a65d5645c2454be95f93dde067658/CBot/src/CBot/CBotDefines.h#L28
robotmain.cpp has notoriously long stretches of barely readable code that should be rewritten, this is just one such fragment.

Reply:
This version filed should **not** be updated "before every single major release". This field should **only** be changed when we break backwards compatibility of the `cbot.run` file or, like in this case, when we can't trust old `cbot.run` files not to crash the game. I'll turn the magic number into a constant.

Reply:
> since it is outside of this PR's scope I'd suggest just leaving the 'version' as is for now.

Nope. This is an essential part of the pull request. The whole point is that we want to be able to change alien AI scripts without causing old save files to crash the game - we have to discard cbot.run.

Reply:
I get the idea, I'm just not fond of the old code.

And I cannot be sure we're going to remember to update it (especially when it's hidden away like this and not at all documented) every time we change something in CBOT and/or the game that would break backwards compatibility. 

Besides, loading old save files is generally unsupported. We more often than not encourage the players to remove their save files after each update. Loading an old save file might work, or it might not - preventing a crash is always great though so I understand your reasoning. However, discarding `cbot.run` would already break some levels anyway so I don't really see the point of using such a save file.

On a related note, we might want to add a warning on loading old save files and actually make use of the part of the code that writes the game version to `data.sav` in the future.

Reply:
> We more often than not encourage the players to remove their save files after each update.

1. No we don't. If it's not in the user interface, it does not count because [uses don't read the documentation](https://www.joelonsoftware.com/2000/04/26/designing-for-people-who-have-better-things-to-do-with-their-lives/).
2. The two most recent release announcements [0.2.1](https://colobot.info/2023/08/08/gotofixes/) and [0.2.0](https://colobot.info/2021/08/21/planb/) did not contain such a warning either.
3. [INSTALL.md has no such warnings eitehr](https://github.com/colobot/colobot/blob/c56d240a0f8a65d5645c2454be95f93dde067658/INSTALL.md)
4. [And neither does README-dev.md](https://github.com/colobot/colobot/blob/c56d240a0f8a65d5645c2454be95f93dde067658/docs/README-dev.md)

> we might want to add a warning on loading old save files

My plan now is to add a new pull request that _prevents_ the game from listing a save file as available to be loaded unless the version of the game that created the file exactly matches the current release. Only supporting save files created by the same game release should make our lives much simpler (e.g. don't need to worry about breaking save files when introducing new built-in functions). We may still need the current pull request too because of mods as well as people play-testing the dev branch.